packages/react-reconciler/src/ReactFiberCommitWork.js

```diff
@@ -1521,6 +1531,16 @@ function attachSuspenseRetryListeners(finishedWork: Fiber) {
         thenable.then(retry, retry);
       }
     });
+  } else {
```

Comment:
As I mentioned above, let's move this out to a separate function that isn't shared with SuspenseList since this is unrelated to attaching listeners.